File: app/src/main/java/com/bnorm/barkeep/data/api/model/Ingredient.java
package com.bnorm.barkeep.data.api.model;

import java.util.List;

import android.os.Parcel;
import android.os.Parcelable;

public class Ingredient implements Parcelable {

    private String name;
    private List<String> nameWords;
    private UnitType type;

    public Ingredient() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<String> getNameWords() {
        return nameWords;
    }

    public void setNameWords(List<String> nameWords) {
        this.nameWords = nameWords;
    }

    public UnitType getType() {
        return type;
    }

    public void setType(UnitType type) {
        this.type = type;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.name);
        dest.writeStringList(this.nameWords);
        dest.writeInt(this.type == null ? -1 : this.type.ordinal());
    }

    protected Ingredient(Parcel in) {
        this.name = in.readString();
        this.nameWords = in.createStringArrayList();
        int tmpType = in.readInt();
        this.type = tmpType == -1 ? null : UnitType.values()[tmpType];
    }

    public static final Creator<Ingredient> CREATOR = new Creator<Ingredient>() {
        public Ingredient createFromParcel(Parcel source) {
            return new Ingredient(source);
        }

        public Ingredient[] newArray(int size) {
            return new Ingredient[size];
        }
    };
}

File: app/src/main/kotlin/com/bnorm/barkeep/data/api/model/Ingredient.kt
package com.bnorm.barkeep.data.api.model

import android.os.Parcel
import android.os.Parcelable

data class Ingredient(var name: String? = null,
                 var nameWords: List<String>? = null,
                 var type: UnitType? = null) : Parcelable {

    companion object {
        @JvmField val CREATOR = createParcel { Ingredient(it) }
    }

    protected constructor(parcel: Parcel) :
            this(parcel.readString(),
                    parcel.createStringArrayList(),
                    parcel.readInt().let { if (it == -1) null else UnitType.values()[it] })

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.name)
        dest.writeStringList(this.nameWords)
        dest.writeInt(if (this.type == null) -1 else this.type!!.ordinal)
    }
}
